extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java

```diff
@@ -308,47 +326,52 @@ public void exceptionOccurred (Throwable exception) {
         return basedir;
     }
 
+	private SimpleFeatureCollection getFeatures(ExtractorLayerRequest request, SimpleFeatureType sourceSchema,
+			SimpleFeatureSource featureSource) throws IOException, TransformException, FactoryException {
+		
+		Query query = createQuery(request, sourceSchema);
+		SimpleFeatureCollection features = featureSource.getFeatures(query);
+		
+		CoordinateReferenceSystem returnedCrs = features.getSchema().getCoordinateReferenceSystem();
+		CoordinateReferenceSystem targetCrs = request._projection;
+		// current version (9.2) of WFS datastore does not perform reprojection
+		if (!CRS.equalsIgnoreMetadata(targetCrs, returnedCrs)) {
+			features = new ReprojectingFeatureCollection(features, targetCrs);
+		}
+		
+		return features;
+	}
+
 	/* This method is default for testing purposes */
     Query createQuery (ExtractorLayerRequest request, FeatureType schema) throws IOException, TransformException,
             FactoryException {
-        switch (request._owsType) {
-        case WFS:
 
-            // bbox may not be in the same projection as the data so it sometimes necessary to reproject the request BBOX
-            ReferencedEnvelope bbox = request._bbox;
-            if (schema.getCoordinateReferenceSystem () != null) {
-                bbox = request._bbox.transform (schema.getCoordinateReferenceSystem (), true, 10);
-            }
-
-            FilterFactory2 filterFactory = CommonFactoryFinder.getFilterFactory2 (GeoTools.getDefaultHints ());
-            String propertyName = schema.getGeometryDescriptor ().getLocalName ();
-            PropertyName geomProperty = filterFactory.property (propertyName);
-            Geometry bboxGeom = new GeometryFactory ().toGeometry (bbox);
-            String epsgCode = "EPSG:"+CRS.lookupEpsgCode(bbox.getCoordinateReferenceSystem(),false);
-            bboxGeom.setUserData(epsgCode);
-
-            Literal geometry = filterFactory.literal (bboxGeom);
-            Intersects filter = filterFactory.intersects (geomProperty, geometry);
-
-            List<String> properties = new ArrayList<String> ();
-            for (PropertyDescriptor desc : schema.getDescriptors ()) {
-                if (desc instanceof GeometryDescriptor && desc != schema.getGeometryDescriptor ()) {
-                    // shapefiles can only have one geometry so skip any
-                    // geometry descriptor that is not the default
-                    continue;
-                } else {
-                    properties.add (desc.getName ().getLocalPart ());
-                }
-            }
-
-            String[] propArray = properties.toArray (new String[properties.size ()]);
-            Query query = new Query (request.getWFSName(), filter, propArray);
-
-            query.setCoordinateSystemReproject (request._projection);
-
-            return query;
-        default:
-            return null;
-        }
+		final CoordinateReferenceSystem nativeCrs = schema.getCoordinateReferenceSystem();
```

Comment:
nativeCrs should be declared in the else part of the indirection

Reply:
done